logging/logrus/grpclogger.go

```diff
@@ -4,12 +4,70 @@
 package grpc_logrus
 
 import (
+	"fmt"
+
 	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc/grpclog"
 )
 
 // ReplaceGrpcLogger sets the given logrus.Logger as a gRPC-level logger.
 // This should be called *before* any other initialization, preferably from init() functions.
 func ReplaceGrpcLogger(logger *logrus.Entry) {
-	grpclog.SetLogger(logger.WithField("system", SystemField))
+	grpclog.SetLoggerV2(&logrusGrpcLoggerV2{logger: logger.WithField("system", SystemField)})
+}
+
+type logrusGrpcLoggerV2 struct {
+	logger *logrus.Entry
```

Comment:
If we embed this instead we can just define the methods that don't already exist on `*logrus.Entry`, i.e., it removes the need to define `Info`,  `Infof`, `Error`, etc.